### PR TITLE
[14] Initialize project code base

### LINKED ISSUE #14: Initialize project code base
Initialize Project codebase using https://github.com/nimblehq/gin-templates

## Acceptance criteria

- The codebase must be available o 2 branches: `main`, `develop`

### PR DESCRIPTION
**Pull request by byhbt**
Ticket: https://github.com/byhbt/go-crawler/issues/14

## What happened 👀

Initialize project codebase using [Gin Template](https://github.com/nimblehq/gin-templates).

Notice: In this PR, I just initialize the code base, not integrate CI, CD yet.

 
## Insight 📝

n/a
 
## Proof Of Work 📹

The source can be up and running locally.

Screen Shot 2021-03-09 at 3 04 43 PM

Screen Shot 2021-03-09 at 3 04 56 PM

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module github.com/nimblehq/default-app-name
```
Comment (hoangmirs) [inline go.mod]: Should we change the `default-app-name` to your project name?

File: config/app.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+app_name = "default-app-name"
```
Comment (hoangmirs) [inline config/app.toml]: Should we change the `default-app-name` to your project name?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Setup Go environment
+        uses: actions/setup-go@v2.1.3
+        with:
+          go-version: 1.16.x
+
+      - name: Create ENV file
```
Comment (hoangmirs) [inline .github/workflows/test.yml]: Should it be `Create .env file`? I'm confused with `Create an environment variable named file`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+https://github.com/nimblehq/git-template/issues/??
```
Comment (hoangmirs) [inline .github/PULL_REQUEST_TEMPLATE.md]: As this is a template for your project, should we change it to your repository here?

File: .github/workflows/test.yml (hunk 3 above)
Comment (byhbt) [reply .github/workflows/test.yml]: yes, I updated it.

File: config/app.toml (hunk 2 above)
Comment (byhbt) [reply config/app.toml]: sure, I also updated other places.

File: .github/workflows/test.yml (hunk 3 above)
Comment (gutakk) [reply .github/workflows/test.yml]: You can also update this in Gin template.

File: .env.example [hunk 5]
```diff
@@ -0,0 +1 @@
+DATABASE_URL=postgres://postgres:postgres@0.0.0.0:5432/default-app-name_development?sslmode=disable
```
Comment (gutakk) [inline .env.example]:
```suggestion
DATABASE_URL=postgres://postgres:postgres@0.0.0.0:5432/go_crawler_development?sslmode=disable
```

**Review by gutakk — COMMENTED** [review]
Please note that Gin template already has CD.  You need to add some secret value that CD needed. Do you plan to work on the separate branch to make the CD pass?

File: .github/workflows/test.yml (hunk 3 above)
Comment (byhbt) [reply .github/workflows/test.yml]: yes, noted, I will create an issue on Gin-Template

File: .env.example (hunk 5 above)
Comment (byhbt) [reply .env.example]: I updated the DB name.

Comment (byhbt) [reply .env.example]: hi @gutakk !

Oh cool, that `Gin Template already has CD`. 
I will update it in another PR, to make the PR clearer.
For this PR I think we just need to initialize the base source code.

**Review by hoangmirs — APPROVED** [review]

**Review by olivierobert — APPROVED** [review]

**Review by gutakk — APPROVED** [review]

**Pull request merged**